library/src/main/java/com/github/lukaspili/reactivebilling/parser/PurchaseTypeParser.java

```diff
@@ -8,12 +8,13 @@
 public class PurchaseTypeParser {
 
     public static PurchaseType parse(String value) {
-        if (value.equals("inapp")) {
-            return PurchaseType.PRODUCT;
-        } else if (value.equals("subs")) {
-            return PurchaseType.SUBSCRIPTION;
-        } else {
-            throw new IllegalArgumentException("Unknown purchase type: " + value);
+        switch (value) {
```

Comment:
This can be dangerous if `value` isn't marked `@NotNull`.  Users will get an NPE if it is.